### PR TITLE
added followers

### PR DESCRIPTION
**Pull request by tubhutani13**

### CODE REVIEW

File: db/migrate/20230120082902_create_follows.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateFollows < ActiveRecord::Migration[7.0]
+  def change
+    create_table :follows do |t|
+      t.integer :follower_id
```
Comment (shubhamjuneja1) [inline db/migrate/20230120082902_create_follows.rb:4]: t.references :follower

File: config/routes.rb [hunk 2]
```diff
@@ -5,10 +5,15 @@
   root "home#index"
   get "/signup", to: "users#new"
   resources :users do
+    get '/followers', to: "users#followers"
+    get '/following', to: "users#followees"
     member do
       get :confirm_email
     end
   end
+  post '/users/:id/follow', to: "users#follow", as: "follow_user"
```
Comment (shubhamjuneja1) [inline config/routes.rb]: should have used  post 'follow', on: :member

File: config/routes.rb [hunk 3]
```diff
@@ -5,10 +5,15 @@
   root "home#index"
   get "/signup", to: "users#new"
   resources :users do
+    get '/followers', to: "users#followers"
```
Comment (shubhamjuneja1) [inline config/routes.rb]: could be simplifed
we don't need this anyways

File: app/views/users/_user.html.erb [hunk 4]
```diff
@@ -5,20 +5,24 @@
...
 </div>
- <div class= "d-flex w-auto  mt-5">
-    <div class="list-group border border-secondary w-50 mr-5">
+<div class= "d-flex w-auto  mt-5">
+  <div class="list-group border border-secondary w-50 mr-5">
     <div class= "border-bottom border-secondary text-center p-2">Your Published Questions</div>
-      <%= render partial: 'questions/question', collection: @user.questions.where.not(published_at: nil) %>
-    </div>
-    <div class="list-group border border-secondary w-50 ml-5">
+    <%= render partial: 'questions/question', collection: @user.questions.where.not(published_at: nil) %>
```
Comment (shubhamjuneja1) [inline app/views/users/_user.html.erb:22]: @user.questions.where.not(published_at: nil) ??

File: app/views/users/_user.html.erb [hunk 5]
```diff
@@ -5,20 +5,24 @@
     <div ><Strong><%= @user.name %></Strong></div>
     <div><%= @user.email %></div>
     <br />
+    <div class= "d-flex justify-content-around">
+      <%= button_to 'Followers', user_followers_path(@user), method: :get, class: "mt-2 btn btn-primary" %>
```
Comment (shubhamjuneja1) [inline app/views/users/_user.html.erb:9]: Please show what's the use

File: app/models/user.rb [hunk 6]
```diff
@@ -9,6 +9,11 @@ class User < ApplicationRecord
   has_many :answers, dependent: :nullify
   has_many :comments, dependent: :nullify
 
+  has_many :followed_users, foreign_key: :follower_id, class_name: "Follow"
+  has_many :followees, through: :followed_users
```
Comment (shubhamjuneja1) [inline app/models/user.rb:18]: can we use has_and_belongs_to_many

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -42,6 +42,18 @@ def confirm_email
     end
   end
 
+  def follow
+    @user = User.find(params[:id])
+    current_user.followees << @user
+    redirect_back(fallback_location: user_path(@user))
+  end
+
+  def unfollow
```
Comment (shubhamjuneja1) [inline app/controllers/users_controller.rb:48]: we could have used single action using js
and toggle follow/unfollow

File: app/models/follow.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class Follow < ApplicationRecord
+    belongs_to :follower, class_name: 'User'
```
Comment (shubhamjuneja1) [inline app/models/follow.rb:2]: do we need this modal?

**Review by shubhamjuneja1 — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 6 above)
Comment (tubhutani13) [reply app/models/user.rb:18]: please discuss

**Pull request merged**